lib/ContextReplacementPlugin.js

```diff
@@ -46,7 +46,7 @@ class ContextReplacementPlugin {
 		const newContentCreateContextMap = this.newContentCreateContextMap;
 
 		compiler.hooks.contextModuleFactory.tap("ContextReplacementPlugin", (cmf) => {
-			cmf.plugin("before-resolve", (result, callback) => {
+			cmf.hooks.beforeResolve.tapAsync("ContextReplacementPlugin", (result, callback) => {
```

Comment:
Use `tap` instead of `tapAsync` and remove the `callback`. Return the result instead.

Reply:
How should I translate `callback(err)` vs `callback(null, result)`?